demo-app/src/androidTest/java/com/worldpay/access/checkout/sample/card/standard/testutil/NewCardFragmentTestUtils.kt

```diff
@@ -0,0 +1,197 @@
+        wait2 { onView(accessCheckoutFieldViewMatcher.viewMatcher).check(matches(accessCheckoutFieldViewMatcher.enabledViewMatcher)) }
+
+    }
+
+    fun clickSubmitButton() = apply {
+        enabledStateIs(submitButton = true)
+        uiObjectWithId(R.id.card_flow_btn_submit).click()
+    }
+
+    fun focusOn(input: Input) = apply {
+        when (input) {
+            Input.PAN -> uiObjectWithId(R.id.card_flow_text_pan).click()
+            Input.CVC -> uiObjectWithId(R.id.cvc_flow_text_cvc).click()
+            Input.EXPIRY_DATE -> uiObjectWithId(R.id.card_flow_expiry_date).click()
+        }
+    }
+
+    fun enterCardDetails(
+        pan: String? = null,
+        cvc: String? = null,
+        expiryDate: String? = null,
+        assertText: Boolean = false
+    ) = apply {
+        if (pan != null) enterText(R.id.card_flow_text_pan, pan)
+        if (cvc != null) enterText(R.id.card_flow_text_cvc, cvc)
+        if (expiryDate != null) enterText(R.id.card_flow_expiry_date, expiryDate)
+
+        if (assertText) {
+            cardDetailsAre(pan, cvc, expiryDate)
+        }
+    }
+
+    fun enterPanCvcExpirydate(pan: String, cvc: String, expiryDate: String) = apply {
+        assertViewIsVisible(R.id.card_flow_text_pan)
+        assertViewIsVisible(R.id.card_flow_text_cvc)
+        assertViewIsVisible(R.id.card_flow_expiry_date)
+        enterTextOnViewWithId(pan, R.id.card_flow_text_pan)
+        enterTextOnViewWithId(cvc, R.id.card_flow_text_cvc)
+        enterTextOnViewWithId(expiryDate, R.id.card_flow_expiry_date)
+    }
+
+    fun clearCardDetails(
+        pan: Boolean? = null,
+        cvc: Boolean? = null,
+        expiryDate: Boolean? = null
+    ) {
+        if (pan == true) clearText(R.id.card_flow_text_pan)
+        if (expiryDate == true) clearText(R.id.card_flow_expiry_date)
+        if (cvc == true) clearText(R.id.card_flow_text_cvc)
+    }
+
+    fun setCursorPositionOnPan(position: Int) = apply {
+        setCursorPosition(R.id.card_flow_text_pan, position, position)
+    }
+
+    fun cursorPositionIs(position: Int) = apply {
+        wait2 { onView(withId(R.id.card_flow_text_pan)).check { view, noViewFoundException ->
+            val isPosition = (view as AccessCheckoutEditText).selectionEnd == position
+            isPosition
+            }
+        }
+    }
+
+    fun cardDetailsAre(pan: String? = null, cvc: String? = null, expiryDate: String? = null) = apply {
+        if (pan != null) wait2 { onView(withText(pan)).check(matches(isDisplayed())) }
+        if (pan != null) wait2 { onView(withText(cvc)).check(matches(isDisplayed())) }
+        if (pan != null) wait2 { onView(withText(expiryDate)).check(matches(isDisplayed())) }
+    }
+
+    fun validationStateIs(
+        pan: Boolean? = null,
+        cvc: Boolean? = null,
+        expiryDate: Boolean? = null
+    ) = apply {
+//        if (pan != null) checkValidationState(R.id.card_flow_text_pan, pan, "pan")
```

Comment:
Looks like it's been left behind, it can be removed now

Reply:
I will remove this